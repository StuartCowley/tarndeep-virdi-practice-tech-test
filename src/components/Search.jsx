import React, { useState } from "react";
import "../styles/Search.css";
import { ReactComponent as SearchIcon } from "../assets/searchIcon.svg";
import { getImages } from "../requests/getImages";

export default function Search({ setSearchResults }) {
  const [value, setValue] = useState("");

  const changeValue = (event) => setValue(event.target.value);

  const handleSearch = async (event) => {
    event.preventDefault();
    const imageResult = await getImages(value);
    return setSearchResults(imageResult);
  };

  return (
    <form className="search-form" onSubmit={handleSearch}>
      <input
        className="search-form__input-text"
        type="text"
        onChange={changeValue}
        value={value}
        placeholder="Search..."
        aria-label="Search"
      />
      <button type="submit" className="search-form__btn" onClick={handleSearch}>
        <SearchIcon />
      </button>
    </form>
  );
}
