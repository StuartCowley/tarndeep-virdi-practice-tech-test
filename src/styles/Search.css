.search-form__input-text {
  max-width: 300px;
  width: 70%;
  text-indent: 10px;
}

.search-form {
  display: flex;
  justify-content: center;
  padding-bottom: 20px;
}

.search-form__btn:active {
  transform: scale(1.1);
}
