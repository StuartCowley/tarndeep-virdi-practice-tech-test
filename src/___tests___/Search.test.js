import React from "react";
import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search Component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Search />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders search input field", () => {
    render(<Search />);
    const searchInput = screen.getByRole("textbox");
    expect(searchInput).toBeInTheDocument();
    expect(searchInput).toHaveClass("search--input-text");
  });
});
