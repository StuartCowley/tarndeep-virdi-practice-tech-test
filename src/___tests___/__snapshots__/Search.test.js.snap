// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      aria-label="Search"
      class="search-form__input-text"
      placeholder="Search..."
      type="text"
      value=""
    />
    <button
      class="search-form__btn"
      type="submit"
    >
      <svg>
        searchIcon.svg
      </svg>
    </button>
  </form>
</DocumentFragment>
`;
